Autopilot/Test/Src/TelemetryManager/Test_TelemetryFSM.cpp

```diff
@@ -0,0 +1,110 @@
+#include <gtest/gtest.h>
+
+#include "telemetryManager.hpp"
+#include "telemetryStateClasses.hpp"
+
+
+using namespace std; 
+using ::testing::Test;
+
+TEST(TelemetryManagerFSM, InitialStateIsInitialMode){
+    /***SETUP***/
+    telemetryManager telemetryMng;
+    /***ASSERTS***/
+    ASSERT_EQ(*(telemetryMng.getCurrentState()), initialMode::getInstance());
+}
+
+TEST(TelemetryManagerFSM, InitialTransitionToGround){
+    /***SETUP***/
+    telemetryManager telemetryMng;
+    /***STEPTHROUGH***/
+    telemetryMng.execute();
+    /***ASSERTS***/
+    ASSERT_EQ(*(telemetryMng.getCurrentState()), groundDataPathMode::getInstance());
+}
+
+TEST(TelemetryManagerFSM, GroundToRegularThroughDataValid){
+    /***SETUP***/
+    telemetryManager telemetryMng;
+    /***DEPENDECIES***/
+    telemetryMng.dataValid=true;
+    telemetryMng.dataError=false;
+    /***STEPTHROUGH***/
+    telemetryMng.execute();
+    telemetryMng.execute();
+    /***ASSERTS***/
+    ASSERT_EQ(*(telemetryMng.getCurrentState()), regularReportMode::getInstance());
+}
+
+TEST(TelemetryManagerFSM, GroundToRegularThroughDataInvalid){
+    /***SETUP***/
+    telemetryManager telemetryMng;
+    /***DEPENDENCIES***/
+    telemetryMng.dataValid=false;
+    /***STEPTHROUGH***/
+    telemetryMng.execute();
+    telemetryMng.execute();
+    /***ASSERTS***/
+    ASSERT_EQ(*(telemetryMng.getCurrentState()), regularReportMode::getInstance());
+}
+
+TEST(TelemetryManagerFSM, GroundToErrorThroughDataValid){
+    /***SETUP***/
+    telemetryManager telemetryMng;
+    /***DEPENDENCIES***/
+    telemetryMng.dataValid=true;
+    telemetryMng.dataError=true;
+    /***STEPTHROUGH***/
+    telemetryMng.execute();
+    telemetryMng.execute();
+    /***ASSERTS***/
+    ASSERT_EQ(*(telemetryMng.getCurrentState()), errorReportMode::getInstance());
+}
+
+TEST(TelemetryManagerFSM, GroundToErrorThroughDataInvalid){
+    /***SETUP***/
+    telemetryManager telemetryMng;
+    /***DEPENDENCIES***/
+    telemetryMng.dataValid=false;
+    /***STEPTHROUGH***/
+    telemetryMng.execute(); //to ground
+    for(int index=0; index<4; index++)
+    {
+        telemetryMng.execute();//to regular report
+        telemetryMng.execute();//to ground
+    }
+    //last increase in counter
+    telemetryMng.execute();//to error report
+    /***ASSERT***/
+    ASSERT_EQ(*(telemetryMng.getCurrentState()), errorReportMode::getInstance());
```

Comment:
Good stuff! Apart from what Dhruv wrote, a couple of notes:

1. We're ditching ASSERT_EQ in favor of EXPECT_EQ. (I need to update the unit test doc). They serve the same purpose, but ASSERT_EQ kills the test as soon as there's a failure while EXPECT_EQ declares error but keps the test going (helps with debuggabillity).
2. The "dependency" section of the unit test is used for setting up fakes and mocks (which you don't need to do in this case). What you've put inside it belongs in the Setup section.